packager/media/formats/mp2t/ts_section_pmt.cc

```diff
@@ -95,7 +96,29 @@ bool TsSectionPmt::ParsePsiSection(BitReader* bit_reader) {
 
     // Read the ES info descriptors.
     // Defined in section 2.6 of ISO-13818.
-    RCHECK(bit_reader->SkipBits(8 * es_info_length));
+    int descriptor_tag;
+    int descriptor_length;
+    char lang[] = { 'u', 'n', 'd', '\0' };
+
+    while (es_info_length) {
+	RCHECK(bit_reader->ReadBits(8, &descriptor_tag));
+	RCHECK(bit_reader->ReadBits(8, &descriptor_length));
+
+	if (descriptor_tag == 10) {
+		RCHECK(bit_reader->ReadBits(8, &lang[0]));	// ISO_639_language_code
+		RCHECK(bit_reader->ReadBits(8, &lang[1]));
+		RCHECK(bit_reader->ReadBits(8, &lang[2]));
+		RCHECK(bit_reader->SkipBits(8));		// audio_type
+
+	}
+	else {
+		RCHECK(bit_reader->SkipBits(8 * descriptor_length));
+	}
+
+	es_info_length -= 2 + descriptor_length;
+    }
```

Comment:
Add instructions to validate that there are no corruptions in ES info descriptors and skip the additional data if there is.

```c++
RCHECK(es_info_length >= 0);
RCHECK(bit_reader->SkipBytes(es_info_length);
```